exercises/difference-of-squares/SquaresSpec.groovy

```diff
@@ -0,0 +1,30 @@
+@Grab('org.spockframework:spock-core:1.0-groovy-2.4')
+import spock.lang.*
+
+class SquaresSpec extends Specification {
```

Comment:
These test are a good chance to show off Spock's parameterised tests with something like this for each one: 
``` groovy
@Unroll
def 'can square the sum of the numbers up to the given number: #num'() {
    expect:
        new Squares(num).squareOfSums() == expected

    where:
        num   || expected
        5     || 225
        10    || 3025
        100   || 25502500
}
```